src/bp/ui-admin/src/telemetry/client.ts

```diff
@@ -0,0 +1,121 @@
+import axios from 'axios'
+import { lang } from 'botpress/shared'
+import _ from 'lodash'
+import moment from 'moment'
+import ms from 'ms'
+import uuid from 'uuid'
+
+import store from '../store'
+
+interface EventPackageInfoType {
+  [key: string]: {
+    timeout: string
+    getPackage: Function
+  }
+}
+
+export const axiosConfig = {
+  baseURL: window.TELEMETRY_URL,
+  headers: {
+    withCredentials: false
+  }
+}
+
+const dataSchemaVersion = '1.0.0'
+
+const telemetrySchemaVersion = '1.0.0'
+
+const pathsInReduxTracked: string[] = ['user.profile.email', 'version.currentVersion', 'license.licensing.isPro']
+
+const eventPackageInfo: EventPackageInfoType = {}
+
+const getEventLock = (event: string) => {
+  const eventLock = window.localStorage.getItem(event)
+
+  if (eventLock !== null) {
+    return JSON.parse(eventLock)
+  }
+
+  return null
+}
+
+const setEventLock = (event: string, timeout?: string) => {
+  const currentTime = moment().valueOf()
+
+  const pkg = eventPackageInfo[event].getPackage()
+
+  const lock = {
+    package: pkg,
+    expiresAt: (timeout ? ms(timeout) : ms(eventPackageInfo[event].timeout)) + currentTime
+  }
```

Comment:
Why do you store the data payload on the local storage ? Is it necessary? Also, your "pkg" placeholder variable is not necessary, you are not using it elsewhere